packages/expo-updates/ios/EXUpdates/AppLauncher/EXUpdatesAppLauncherWithDatabase.m

```diff
@@ -117,6 +117,24 @@ - (void)launchUpdateWithSelectionPolicy:(EXUpdatesSelectionPolicy *)selectionPol
   }
 }
 
++ (NSArray<NSUUID *> *)storedUpdateIdsInDatabase:(EXUpdatesDatabase *)database
+                                           error:(EXUpdatesAppLauncherErrorBlock)errorBlock
+{
+  __block NSArray<NSUUID *> *readyUpdateIds = @[];
+  dispatch_sync(database.databaseQueue,^{
```

Comment:
We have generally tried to avoid using `dispatch_sync` in non-test code to avoid the risk of UI deadlocks, as you rightly pointed out earlier; I'd probably opt for `dispatch_async` here and a callback signature for this method, but if you'd rather not do that, it might be good to note this in the docblock in `EXUpdatesExternalInterface` so that callers will know to account for this possibility themselves. (Sorry I didn't notice this before!)